administrator/components/com_categories/models/category.php

```diff
@@ -439,7 +440,10 @@ protected function preprocessForm(JForm $form, $data, $group = 'content')
 
 			foreach ($languages as $tag => $language)
 			{
-				if (empty($data->language) || $tag != $data->language)
+				$permission = $user->authorise('language.permission', 'com_languages.language.' . (int) $language->lang_id);
+				$canDoAssociations = !is_null($permission) ? $permission : true;
```

Comment:
if is null i think it meand not allowed so it can't.
So, it should be just this line:
```php
$canDoAssociations = $user->authorise('language.permission', 'com_languages.language.' . (int) $language->lang_id);
```